tests/dataflow/test_unified_systolic.py

```diff
@@ -0,0 +1,121 @@
+# Copyright Allo authors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import allo
+from allo.ir.types import int8, int32, bool, Stream
+import allo.dataflow as df
+import allo.backend.hls as hls
+import numpy as np
+import pytest
+
+M, N, K = 4, 4, 4
+Rt, Ct = 2, 2
+# Rt, Ct = 4, 4
+# M, N, K = 16, 16, 16
+# Rt, Ct = 4, 4
+# Rt, Ct = 16, 16
+P0, P1 = Rt + 2, Ct + 2
+
+
+@df.kernel(mapping=[P0, P1])
+def gemm(A: int32[M, K], B: int32[K, N], inst: int8[2], C: int32[M, N]):
+
+    flowtag: bool = inst[0]
+    # --------------------------------------------------------
+    Rtimes: int32 = M // Rt if flowtag else K // Rt
+    Ctimes: int32 = N // Ct
+    Tlength: int32 = K if flowtag else M
+
+    local_S: int32
+
+    # --------------------------------------------------------
+    # Interconnect
+    i, j = df.get_pid()
+    in_R: Stream[int32] = df.pipe(src=(i, j - 1), dst=(i, j))
+    in_C: Stream[int32] = df.pipe(src=(i - 1, j), dst=(i, j))
+    out_R: Stream[int32] = df.pipe(src=(i, j), dst=(i, j + 1))
+    out_C: Stream[int32] = df.pipe(src=(i, j), dst=(i + 1, j))
+
+    # --------------------------------------------------------
+    # Computation
+    for ri in range(Rtimes, name="row_loop"):
+        for ci in range(Ctimes, name="column_loop"):
+            # corner
+            with allo.meta_if(i in [0, Rt + 1] and j in [0, Ct + 1]):
```

Comment:
This statement is a bit confusing. For example, for `i in [0, Rt + 1]`, are you saying `i` is either `0` or `Rt+1`? In this case, it would be better to write it as a set (to be consistent with the math notation), i.e., `i in {0, Rt+1}`.